monitor.go

```diff
@@ -37,18 +44,33 @@ func (m *defaultMonitor) Run(c *exec.Cmd) error {
 	return c.Run()
 }
 
-func (m *defaultMonitor) Start(c *exec.Cmd) error {
-	return c.Start()
-}
-
-func (m *defaultMonitor) Wait(c *exec.Cmd) (int, error) {
-	if err := c.Wait(); err != nil {
-		if exitErr, ok := err.(*exec.ExitError); ok {
-			if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
-				return status.ExitStatus(), nil
+func (m *defaultMonitor) Start(c *exec.Cmd) (chan Exit, error) {
+	ec := make(chan Exit, 1)
+	if err := c.Start(); err != nil {
+		close(ec)
+		return nil, err
+	}
+	go func() {
+		var status int
+		if err := c.Wait(); err != nil {
+			status = 255
+			if exitErr, ok := err.(*exec.ExitError); ok {
+				if ws, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+					status = ws.ExitStatus()
+				}
 			}
 		}
-		return -1, err
-	}
-	return 0, nil
+		ec <- Exit{
+			Timestamp: time.Now(),
```

Comment:
nit: get the time as soon as `c.Wait()` returns for more accuracy?

Reply:
i don't think you will get any closer with the error handling

Reply:
Mostly because I don't trust the scheduler :)

Reply:
its sends the value so the fields are all done